File: content/LLD/Reference-Code/22. Ride Sharing Project - Good Example/driver.py
from user import User
from vehicle import Vehicle


class Driver(User):
    def __init__(self, name, email, location, vehicle: Vehicle):
        super().__init__(name, email, location)
        self.__vehicle = vehicle

    def get_vehicle(self) -> Vehicle:
        return self.__vehicle

    def notify(self, msg: str):
        print(f"Notify to driver({self.name}) = {msg}")

File: content/LLD/Reference-Code/22. Ride Sharing Project - Good Example/Driver.java
public class Driver extends User {
    private final Vehicle vehicle;

    public Driver(String name, String email, Location location, Vehicle vehicle) {
        super(name, email, location);
        this.vehicle = vehicle;
    }

    public Vehicle getVehicle() {
        return this.vehicle;
    }

    @Override
    public void notify(String msg) {
        System.out.println("Notify to driver(" + this.name + ") = " + msg);
    }
}
